wordpress-mobile/WordPress-Android
# Issue/13329 backup wizard skeleton

Parent #13329 

This PR supports using the `WizardManager` for the backup download process and includes the basics for getting this started: dagger support, defining steps and simple navigation. The remainder of the logic will be implemented in future PRs.
This PR also includes some cleanup based on Kotlin guidelines - removal of redundant qualifiers

~~NOTE: Lint is failing on an unrelated issue with Utils package. A PR is underway to suppress the error in lint-baseline.xml~~

To test:
- Launch the app and enable the BackupFeatureConfig (restart the app)
- Navigate back to Activity Log
- Tap on more menu
- Tap the "Download backup" menu item
Note that the backup detail view is launched (The feature itself is not functional, this is just to test that the view shows)

PR submission checklist:
